List user references

mrkvon opened the pull request:
#### Proposed Changes

* Read references of user.
React component.

`/profile/{username}/references`
- [x] show list of references which user received
  - [x] name, username, avatar, interactions, recommendation, when the reference was given
  - [ ] design
- [ ] when looking at my references, show which ones I haven't given back (public or private) and how much time is left till they become public. Provide a button.
  - [x] show unpublished references to me
    - [x] avatar, link, time left to give
    - [ ] design
- [ ] show list of references which user gave
- [ ] when looking at my references, show which ones I gave and they're still private (and time left)

#### Testing Instructions

* manual testing

Continues fixing #718 
Depends on #827 
It will need a UX and looks design.

#### To do
- [ ] show a number of references in user tab (separated by recommend (yes,no,unknown))
- [ ] show a list of references when the tab is clicked

### Screenshots

![05_list_references_self](https://user-images.githubusercontent.com/7449720/72669349-ea2ed380-3a30-11ea-9780-a3835f8b12c3.png)

simison commented:
FYI, I'm looking into designs for this.

petr-hajek commented:
> FYI, I'm looking into designs for this.

Still looking into it? Or I could try to make something up.

simison commented:
>  Still looking into it? Or I could try to make something up.

Still actively thinking it but haven't actually started plotting any designs yet. Happy if you get started with it and we can iterate on it together!

petr-hajek commented:
> > Still looking into it? Or I could try to make something up.
> 
> @petr-hajek I'm still actively thinking it but haven't actually started plotting any designs yet. Happy if you get started with it and we can iterate on it together!

I will be going out of the collective soon, so I wanna wrap things up. I'll be probably offline for longer period now.

simison commented:
This PR will need same shallow `setState` changes as https://github.com/Trustroots/trustroots/pull/827

It's okay if we first get very rough version of this merged. We can then work out the UI / design out before publishing the feature.

mrkvon commented:
A raw displaying of references on the reference page is more or less done. We'll need a design to progress further.

mrkvon commented:
Hello there! I need a design or a code review by someone to progress on this further.
I'll self-review today.
Merge without design would also be an option (and then it may stay hidden behind flags and be tested and iterated upon on a dev server or something (dev server is down though)).

simison commented:
@mrkvon thanks for the nudge. I'd say go ahead with self-review, I'll loop back to design aspects.

Let's chat more later about speeding up the process and freeing you up from reviews a bit, going forward. :-)

simison commented:
Rebased! :tada:

simison commented:
Could someone take screengrabs of the feature as it is currently working here? It would be helpful for me to see and give feedback.

mrkvon commented:
> take screengrabs

@simison a screenshot is in the PR description

Additional screenshots (create a reference) will appear in #718 shortly.

Also rebased.

mrkvon commented:
Rebased and fixed conflicts. Anyways, this is not in my focus, currently. Feel free to close or take over this PR.

mrkvon commented:
Rebased this with @edvardmk and @rumwerfer. Also refactored a little.

Who wants to implement listing references? This PR is for grabs.

I'm ready to answer questions about this PR and give support to whoever may want to continue it.
Alternatively, if [somebody provides a clear specification based on consensus and at least a basic user interface design] and [nobody else is interested in working on this] and [this is needed], then i may implement it myself.

Also depending on what we want it may be easier to close this and start from scratch.

Akronix commented:
hey @mrkvon I'm up to take some time slots of my agenda this week and work to polish this.

simison commented on modules/core/client/components/Loading.js:
```diff
@@ -0,0 +1,10 @@
+/**
+ * I'm just a little loading info.
+ */
+export default function Loading() {
```
Let's use existing component:

https://github.com/Trustroots/trustroots/blob/master/modules/core/client/components/LoadingIndicator.js

simison commented on modules/references/client/components/create-reference/Recommend.js:
```diff
@@ -17,13 +17,15 @@ export default function Recommend({
 
   const question = recommendQuestions[primaryInteraction];
 
+  const marginRight = { marginRight: '20px' };
```
Let's use either styled components in JS or CSS in less files, but no inline CSS via JS.

Akronix replied:
OK. The plan is to change upgrade react-bootstrap lib anyway and use different styling options from the new version.

Akronix replied:
Is it resolved now?

mrkvon commented:
I'd be in favor of opening a separate pull request based on the references-private-experience branch and merging that to master. We don't need to care about (or merge) all the #918 stuff when we want the UX changes in reference creating experience. That is, the recent changes by @Akronix and @rumwerfer.

Akronix commented:
I agree. Let's close this to avoid further confusion

nicksellen commented:
^ I just restored the branch in case there is some useful code in there to reuse for the eventual branch we create...
